src/coreComponents/python/modules/pygeosx_tools_package/pygeosx_tools/mesh_interpolation.py

```diff
@@ -0,0 +1,202 @@
+    """
+    Get random realizations for noisy signals on target bins
+
+    Args:
+        x (np.ndarray): 1D list/array of positions
+        bins (np.ndarray): 1D list/array of bin edges
+        targets (dict): Dict of geosx target keys, inputs to get_random_realization
+
+    Returns:
+        dict: Dictionary of random realizations
+
+    """
+    results = {}
+    for ka in targets.keys():
+        results[ka] = get_random_realization(x, bins, **targets[ka])
+    return results
+
+
+def get_surface_interpolator(fname, z_origin=2700.0):
+    """
+    Parse a surface definition file
+
+    Args:
+        fname (str): Path of the target file
+        z_origin (float): Vertical offset of a target mesh file (default=2700.0)
+
+    Returns:
+        tuple: Surface interpolators (linear-depth, nearest-depth, linear-x, nearest-x)
+    """
+    nodes = []
+    with open(fname, 'r') as f:
+        for line in f:
+            if ('#' not in line):
+                tmp = line.split(',')
+                if ('.' in line):
+                    nodes.append([float(x) for x in tmp])
+
+    # Convert values
+    nodes = np.array(nodes)
+    x = np.ascontiguousarray(np.squeeze(nodes[:, 2])) / 3.281
+    y = np.ascontiguousarray(np.squeeze(nodes[:, 0])) / 3.281
+    z = (np.ascontiguousarray(np.squeeze(nodes[:, 1])) - z_origin) / -3.281
+
+    # Generate four interpolators to reconstruct the surface
+    interp_a = interpolate.LinearNDInterpolator((x, y), z)
+    interp_b = interpolate.NearestNDInterpolator((x, y), z)
+    interp_c = interpolate.NearestNDInterpolator((x, y), x)
+    interp_d = interpolate.NearestNDInterpolator((x, y), y)
+
+    return interp_a, interp_b, interp_c, interp_d
+
+
+def build_conforming_tables(property_interps, surface_interps, axes, extrapolate_surfaces=True):
+    """
+    Build a set of conforming tables property, surface interpolators
+
+    Args:
+        property_interps a dict of 1D property interpolators with depth
+        surface_interps a dict of interpolators for surfaces
+        axes a list of x,y,z axis definitions
+        extrapolate_surfaces a flag to indicate whether surface elevations should be extrapolated (default=True)
+
+    Returns:
+        tuple: Conforming tables, surface elevations at grid, grid values
+    """
+    # Sort the surfaces by depth at the origin
+    surfaces = list(surface_interps.keys())
+    z_pilot = [surface_interps[ka][1](0.0, 0.0) for ka in surfaces]
+    Ia = np.argsort(z_pilot)
+    z_pilot = np.array(z_pilot)[Ia]
+    surfaces = np.array(surfaces)[Ia]
+
+    # Interpolate values onto a new grid
+    x_grid, y_grid, z_grid = axes
+    zo = np.concatenate([[-1e6], z_pilot, [0.0]], axis=0)
+    X, Y, Z = np.meshgrid(x_grid, y_grid, z_grid, indexing='ij')
+    conforming_tables = {k: np.zeros(np.shape(X)) for k in property_interps.keys()}
+    surface_elevations = np.zeros((len(x_grid), len(y_grid), len(surfaces)))
+
+    for ii in range(len(x_grid)):
```

Comment:
```suggestion
    for ii, xx in enumerate(x_grid):
```

Reply:
This function was specific to a problem I was working on and probably shouldn't belong in the repository